app/src/main/java/arctouch/upcomingmovies/fragments/MainFragment.java

```diff
@@ -0,0 +1,183 @@
+ * An ASyncTask is used to make a GET request from the server.
+ *
+ * @author Cassio Espindola
+ */
+public class MainFragment extends Fragment {
+    /**
+     * List which will contains all the movies fetched from the server.
+     */
+    private ArrayList<Movie> movies;
+    /**
+     * Progress dialog instance.
+     */
+    private ProgressDialog pDialog;
+    /**
+     * Movie list adapter.
+     */
+    private MovieAdapter adapter;
+    /**
+     * Represents the result page. Each page contains 20 results.
+     */
+    private int page = 1;
+
+    @Override
+    public View onCreateView(final LayoutInflater inflater, final ViewGroup container,
+                             final Bundle savedInstanceState) {
+
+        View rootView = inflater.inflate(R.layout.fragment_main, container, false);
+
+        //Initializing the main list.
+        ListView listOfMovies = rootView.findViewById(R.id.movies_list_view);
+
+        // Creating the list view footer button "Load More".
+        Button btnLoadMore = new Button(getActivity());
+        btnLoadMore.setText(getString(R.string.load_more));
+        btnLoadMore.setTextColor(getResources().getColor(android.R.color.white));
+        listOfMovies.addFooterView(btnLoadMore);
+        btnLoadMore.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                //When "Load More" button clicked, request the next page of results.
+                page++;
+                new GetRequest().execute();
+            }
+        });
+
+        //List of Movies used by adapter.
+        movies = new ArrayList<Movie>();
+        adapter = new MovieAdapter(getActivity(), R.layout.row, movies);
+        listOfMovies.setAdapter(adapter);
+        listOfMovies.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+                android.support.v4.app.FragmentTransaction ft = getActivity().getSupportFragmentManager().beginTransaction();
+                ft.setTransition(FragmentTransaction.TRANSIT_FRAGMENT_OPEN);
+                MovieFragment movieFragment = new MovieFragment();
+
+                Bundle bundle = new Bundle();
+                Movie movie = adapter.getItem(i);
+                bundle.putSerializable("movie", movie);
+                movieFragment.setArguments(bundle);
+                ft.replace(android.R.id.content, movieFragment);
+                ft.addToBackStack(null);
+                ft.commit();
+            }
+        });
+
+        // Calls ASyncTask to make a GET request to the server.
+        new GetRequest().execute();
+
+        // Inflate the layout for this fragment.
+        return rootView;
+    }
+
+    /**
+     * This private class is responsible to make a request to the server.
+     */
+    private class GetRequest extends AsyncTask<Void, Void, Void> {
```

Comment:
Considering good standards to architecture an Android project and also thinking in Object Oriented aspects, Do you think this is the best place to keep the code of this inner class that is requesting movies from the Api?

Reply:
I think it could be better to put this code in a separated class to preserve encapsulation and abstraction.